daemon/cluster/listen_addr.go

```diff
@@ -223,14 +223,16 @@ ifaceLoop:
 		// and exactly one IPv6 address, favor IPv4 over IPv6.
 		if interfaceAddr4 != nil {
 			if systemAddr != nil {
-				return nil, errMultipleIPs
+				return nil, fmt.Errorf("could not choose an IP address to advertise since this system has multiple addresses (%s on %s and %s on %s) on different interfaces", systemAddr, systemInterface.Name, interfaceAddr4, intf.Name)
```

Comment:
Minor suggestion: Move the addresses and interfaces after "on different interfaces".

`could not choose an IP address to advertise since this system has multiple addresses on different interfaces (%s on %s and %s on %s)`